src/Gjallarhorn/CoreTypes.fs

```diff
@@ -265,6 +265,8 @@ type internal ObservableToSignal<'a when 'a : equality>(valueProvider : IObserva
 
     /// Default implementations work off single set of dependenices
     member __.HasDependencies with get() = dependencies.HasDependencies
+    member private _.SetDependencies(deps:IDependencyManager<'a>) = dependencies <- deps
+    member private _.SetWeakSubscription(subs:IDisposable)= weakSubscription <- subs
```

Comment:
Since these really must always be called , and called in order, can we make this a single method to handle init?